TicTacToe/Program.cs

```diff
@@ -9,238 +10,292 @@ namespace TicTacToe
-                    Console.SetCursorPosition(42 + j * 4, 14 + i * 4);
-                    Console.Write(board[9][location]);
+                    else
+                    {
+                        Console.ForegroundColor = ConsoleColor.DarkRed;
+                    }
+                    Console.SetCursorPosition(i, j);
+                    Console.Write("#");
 
                 }
             }
 
-        }
-        private static void buildBoard()
-        {
-            for (int i = 0; i <= 36; i += 4)
+            for (int j = y; j <= y + distanceBoards; j += spaceBetween)
             {
-                for (int j = 0; j <= 36; j++)
+                for (int i = x; i <= x + distanceBoards; i ++)
                 {
-                    if (i % 12 == 0 || j % 12 == 0)
+                    if (i <= distanceBoards * innerCells && i % distanceBoards == 0 || i <= distanceBoards * innerCells && j % distanceBoards == 0)
                     {
                         Console.ForegroundColor = ConsoleColor.Red;
-                        Console.SetCursorPosition(i, j);
-                        Console.Write("#");
-                        Console.SetCursorPosition(j, i);
-                        Console.Write("#");
                     }
                     else
                     {
                         Console.ForegroundColor = ConsoleColor.DarkRed;
-                        Console.SetCursorPosition(i, j);
-                        Console.Write("#");
-                        Console.SetCursorPosition(j, i);
-                        Console.Write("#");
                     }
-                }
-            }
-            for (int i = 40; i < 53; i += 4)
-            {
-                for (int j = 12; j < 25; j++)
-                {
-                    Console.ForegroundColor = ConsoleColor.Red;
                     Console.SetCursorPosition(i, j);
                     Console.Write("#");
                 }
             }
-            for (int i = 12; i < 25; i += 4)
+
+            // Updating the game board
+            int z = 0;
+            for (int j = y + startPosition; j <= y + distanceBoards - startPosition; j += spaceBetween)
             {
-                for (int j = 40; j < 53; j++)
+                for (int i = x + startPosition; i <= x + distanceBoards - startPosition; i += spaceBetween)
                 {
-                    Console.ForegroundColor = ConsoleColor.Red;
-                    Console.SetCursorPosition(j, i);
-                    Console.Write("#");
+                    Console.ForegroundColor = ConsoleColor.Yellow;
+                    Console.SetCursorPosition(i, j);
+                    Console.Write(board[smallBoard][z]);
+                    z++;                  
                 }
             }
+
         }
 
-        private static int humanTurn(char[][] board, char player1)
+        private static void BuildBoard(char[][] board)
+        {
+            PrintBoard(0, 0, board, 0);
```

Comment:
Reuse: instead of seeing numbers 0,1,2,3,... hard coded, it's better to have a variable that increments every time it's used.

Tip: you can increment that variable in same line it's used, e.g. first line would be:

`PrintBoard(0,0,board, <variable>++);`

Do you know the difference between above line and writing below line?

`PrintBoard(0,0,board, ++<variable>);`